### PR TITLE
No guest access allowed? Not logged in? No contentnode API calls for you!

### LINKED ISSUE #7314: Learn -> Recommended makes API call regardless of auth
### Observed behavior

I went to make sure that my device settings prohibiting access to Learn for guests was working properly.

The "Channels" tab felt like it took longer to load and show the auth message than I expected. So I tried Recommended and I saw in my console that it was making this request on Channels and hitting `/api/content/contentnode/popular` on Recommended.

```
/api/content/contentnode/?ids=ec164fee25ee526296e68f7c10b1e169%2Cd824828e67b9472a9987a96870729178%2C91b4ce67480b51e699c52ae10b00c724%2Ca9b25ac9814742c883ce1b0579448337%2C878ec2e6f88c5c268b1be6f202833cd4%2Cfc47aee82e0153e2a30197d3fdee1128%2C1ceff53605e55bef987d88e0908658c5%2C1d8f6d84618153c18c695d85074952a7%2C913efe9f14c65cb1b23402f21f056e99%2C61b75af2bb2c4c0ea850d85dcf88d0fd%2C27bb0abc24d44dd9896be50d47b2357e%2C09ee940e106953a2b6716e1020a0ce3f%2C000409f81dbe5d1ba67101cb9fed4530%2C277ebe3f6d434b528e45bb8f0ab01607%2C0a3446937e3340fa86e6010ba80e16e1%2Cf2ed9568a9da4157b283f76406df472c%2C1c98e92b8c2f536796960bed8d137a25%2C95a52b386f2c485cb97dd60901674a98%2C2f3f836672a958ef92097b5e4a9b2ed3%2C737804b26629425284ef48fa1b1e1bde%2C9e1bb851d3f64c338113ec23809da56f%2Cc74b5be808354f318471ba46c68c0aae&user_kind=anonymous&contentCacheKey=1594330842 HTTP/1.1" 200 12001
```

No need to get content if we can't show it to anybody.

### Expected behavior

Avoid unnecessary API calls.

### User-facing consequences

Long loading time for auth message when accessing Learn without authorization.

Unnecessary API calls could affect the server negatively

### Steps to reproduce

Device -> Settings -> then set it so learners must log in to see content in Learn.

Go to `/en/learn` and `/en/learn#/recommended` and take a look at your network tab in Chrome

### Context

Debian 10
0.14.x-beta9
Chrome

### PR DESCRIPTION
**Pull request by nucleogenesis**
### Summary

Fixes https://github.com/learningequality/kolibri/issues/7314

Adds and applies guard to router in Learn that avoids entering any of the route handlers before finding out that you actually don't belong here and sending you to the 404 "please sign in" CONTENT_UNAVAILABLE page.

### Reviewer guidance

When your device setting is to disallow guest access and you're not logged in, do you see any API calls for content?

Also another smoke test of various Learn routes (like if you have classes, lessons, exams - what happens if you use a link directly to them in a new tab?) for some additional assurance would be welcome.

### References

https://github.com/learningequality/kolibri/issues/7314

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/modules/pluginModule.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ export default {
     examLog: {},
     memberships: [],
     canAccessUnassignedContentSetting: plugin_data.allowLearnerUnassignedResourceAccess,
+    allowGuestAccess: plugin_data.allowGuestAccess,
```
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/modules/pluginModule.js:23]: That's weird - we were already bootstrapping this into the learn page, but somehow we don't seem to be using it anywhere before this PR?

File: kolibri/plugins/learn/assets/src/routes/index.js [hunk 2]
```diff
@@ -28,12 +28,35 @@ function unassignedContentGuard() {
   return;
 }
 
+function allowGuestAccessGuard() {
```
Comment (rtibbles) [inline kolibri/plugins/learn/assets/src/routes/index.js]: We need to do this for all the class routes too, right?

Might be best just to do this as a global guard on the router itself?

We do this for `afterEach` here: https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/plugins/learn/assets/src/app.js#L23

So could add `beforeEach` for this check in the same place.

Noting that I didn't do that for the content unavailable, as the behaviour is differential between class and non-class routes.

**Review by rtibbles — COMMENTED** [review]
One possible simplification using a global before guard: https://router.vuejs.org/guide/advanced/navigation-guards.html#global-before-guards

File: kolibri/plugins/learn/assets/src/modules/pluginModule.js (hunk 1 above)
Comment (nucleogenesis) [reply kolibri/plugins/learn/assets/src/modules/pluginModule.js:23]: Yeah I thought that was odd, but appreciated it being there :-D

**Comment by nucleogenesis** [comment]
@rtibbles 

I ran into issues with using a `beforeEach` on my first pass and moved along but after giving it a fresh look I worked it out.

Force pushed a completely different set of changes.

**Review by rtibbles — APPROVED** [review]
Code looks good, manual test checks out.

**Comment by rtibbles** [comment]
Build issue is due to unrelated pipeline configuration.

**Pull request merged**